src/norm/postgres.nim

```diff
@@ -101,6 +101,20 @@ proc createTables*[T: Model](dbConn; obj: T) =
       fkGroups.add "FOREIGN KEY($#) REFERENCES $#($#)" %
         [obj.col(fld), typeof(get val.model).table, typeof(get val.model).col("id")]
 
+    when obj.dot(fld).hasCustomPragma(fk):
+      # Check val is int
+      when val is not int:
+        {.error: "Pragma fk must be used on an integer field. " & fld & " is not an integer." .}
```

Comment:
@Clonkk I've read the manual on those pragmas. They generate compile-time messages but the compilation goes through. Is that the desired behavior? I mean, won't it it be better to crash on invalid model? It's an honest question, I don't think there is an obvious right answer.

Reply:
I did wondered if ``{.fatal: ... .}`` would have been better; in practice, I have observed the behaviour to be the same. Nonetheless, If we want to be 100% sure, I believe ``fatal`` pragma would be more appropriate.